packages/babel-plugin-minify-guarded-expressions/src/index.js

```diff
@@ -17,14 +17,24 @@ module.exports = function({ types: t }) {
 
             const left = path.get("left");
             const right = path.get("right");
+
+            // issues - 171, 174, 176
+            // we assume that it is returned/assigned/part of a bigger expression
+            // or utilized somehow
+            // we check if we shouldBail only when evaluating
+            // the rightside of the expression;
+            // if the left side is evaluated to be deterministic,
+            // we can safely replace the entire expression
+            const shouldBail = !path.parentPath.isExpressionStatement();
```

Comment:
I hope this assumption doesn't have any corner cases that fail. Will add test cases once this seems to work out well for all corner cases.